src/dinit.cc

```diff
@@ -453,6 +471,89 @@ static int process_commandline_arg(char **argv, int argc, int &i, options &opts)
     return 0;
 }
 
+#if SUPPORT_SELINUX
+// Load the system SELinux policy and transition ourselves to it.
+// First, load the policy with selinux_init_load_policy(3). We don't
+// need to worry about the enforcing=0 kernel cmdline option or parsing
+// /etc/selinux/config, selinux_init_load_policy(3) will handle all cases
+// for us.
+//
+// This should be done as early as possible. We want to transition the init
+// system to the domain specified by the policy as early as possible for security
+// reasons. On top of this, we need to ensure we don't have open fd's lying about
+// before we transition, as they'll get stuck effectively having the old context
+// dinit initally started with before loading the policy.
+//
+// We take a parameter, const char *exe, which is argv[0], i.e. the path that
+// we are invoked with, to calculate our new security context to tranition
+// into.
+//
+// If we fail to load the system SELinux policy, return false, otherwise,
+// return true.
```

Comment:
I recommend updating this comment to more focus on what selinux_transition() does and less on how it's expected to be used.

Reply:
I would say: it's ok to detail how it's expected to be used, but I agree there should be more detail on what it is for, and preferably expressed less in terms that require knowledge of SELinux.

I'm thinking along the lines of the following (and this is from my own limited understanding so may need to be adjusted, but just to give an idea):

> Load the SELinux policy. When successful this will cause SELinux labels to be attached to processes and file descriptors and the SELinux framework will begin to enforce restrictions on access based on these labels and the loaded policy.
>
> We might lose access to any file descriptors we have open when this is called (since they will still be labelled with the kernel context), so it is best done early (i.e. before we start opening file descriptors).

Overall the comment should follow the style from the CODE-STYLE:

```
   8.3 Comment structure for functions: short description, parameters, return value, and any
       exceptions that may be thrown (with reason the exception may be thrown, if it is not
       implicit in the type). Any longer/detailed notes can follow after the above.
       
       Example:
       
       // Add a dependency. Caller must ensure that the services are in an appropriate state and
       // that a circular dependency chain is not created. Propagation queues should be processed
       // after calling this (if dependency may be required to start).
       // Parameters:
       //   to - target of dependency to be added
       //   dep_type - the type of the dependency
       //   i - where to insert the dependency (in dependencies list)
       // Returns:
       //   A reference to the dependency just added
       // Throws:
       //   std::bad_alloc

```

(I realise a lot of existing comments don't exactly match that style yet, but it is what I'm working towards).

I hope that's clear.

Reply:
Done. I've made some small changes to that, such as changing the wording in this sentence:
```
When successful this will cause SELinux labels to be attached to processes and file descriptors
```
to:
```
When successful, this will cause SELinux labels as per the policy to be attached to processes (and file descriptors owned by those processes).
```
to make it a little more clear that contexts with those file descriptors are more contexts of objects of the overall process.